core/src/main/java/org/apache/carbondata/core/index/IndexUtil.java

```diff
@@ -198,25 +198,17 @@ public static void loadIndexes(CarbonTable carbonTable, IndexExprWrapper indexEx
       String clsName = "org.apache.spark.sql.secondaryindex.Jobs.SparkBlockletIndexLoaderJob";
       IndexJob indexJob = (IndexJob) createIndexJob(clsName);
       String className = "org.apache.spark.sql.secondaryindex.Jobs.BlockletIndexInputFormat";
-      SegmentStatusManager.ValidAndInvalidSegmentsInfo validAndInvalidSegmentsInfo =
-          getValidAndInvalidSegments(carbonTable, FileFactory.getConfiguration());
-      List<Segment> invalidSegments = validAndInvalidSegmentsInfo.getInvalidSegments();
       FileInputFormat indexFormat =
-          createIndexJob(carbonTable, indexExprWrapper, validSegments, invalidSegments,
-              partitionsToPrune, className, false);
+          createIndexJob(carbonTable, indexExprWrapper, validSegments, className);
```

Comment:
Please add an UT with Carbon load indexes parallel enabled, to avoid these issues in future

Reply:
Added a testcase